File: sdk/dotnet/Mics.Client/WsUriBuilder.cs
namespace Mics.Client;

internal static class WsUriBuilder
{
    public static Uri Build(Uri baseUrl, string tenantId, string token, string deviceId)
    {
        ArgumentNullException.ThrowIfNull(baseUrl);
        tenantId ??= "";
        token ??= "";
        deviceId ??= "";

        var builder = new UriBuilder(baseUrl);
        var query = builder.Query;
        if (!string.IsNullOrEmpty(query))
        {
            query = query.TrimStart('?');
            if (query.Length > 0 && !query.EndsWith('&'))
            {
                query += "&";
            }
        }
        else
        {
            query = "";
        }

        query += "tenantId=" + Uri.EscapeDataString(tenantId)
                 + "&token=" + Uri.EscapeDataString(token)
                 + "&deviceId=" + Uri.EscapeDataString(deviceId);

        builder.Query = query;
        return builder.Uri;
    }
}

File: sdk/flutter/mics_client_sdk/lib/src/ws_uri_builder.dart
Uri buildWsUri(String baseUrl, String tenantId, String token, String deviceId) {
  final uri = Uri.parse(baseUrl);
  final qp = Map<String, String>.from(uri.queryParameters);
  qp['tenantId'] = tenantId;
  qp['token'] = token;
  qp['deviceId'] = deviceId;
  return uri.replace(queryParameters: qp);
}

